HW_7/logger.py

```diff
@@ -0,0 +1,11 @@
+from datetime import datetime
+
+
+def log(func):
+    def wapper(*args):
+        with open('log.csv', 'a', encoding='utf-8') as log_file:
+            log_file.write(f'{datetime.now()} - запущена функция' 
+                            '{func.__name__} {"с аргументами" if args else "без аргументов"}' 
+                            '({func.__doc__})\n')
+        return func(*args)
+    return 
```

Comment:
Артем, если вы списываете, то списывайте до конца.
```suggestion
    return wapper
```
Именно из за этого маленького недочеты у вас вся программа и не запускается. Я знаю, что я вам не давал декораторы и в лекциях у вас тоже этого не было. Поэтому нужно было просто переписать это их семинара.

Reply:
Я всё списывал с семинара. Сложно, для меня это очень сложно. С# мне легче давался  Мало знаний, мало навыков, за тобой период обучения, такое не реально просто написать. Я понимаю что дело не в вас, вы даёте информацию в том объеме в котором можете. Но я не понимаю.

Reply:
Вы как специалист, подскажите как, где мне набраться знаний? Может посоветуете книги, какие то обучающие видео может каких то ещё преподавателей ? Нам по с# кучу дополнительного материала давали всякие книги, ссылки на доп видео, просто заваливали материалами.

Reply:
> Вы как специалист, подскажите как, где мне набраться знаний? Может посоветуете книги, какие то обучающие видео может каких то ещё преподавателей ? Нам по с# кучу дополнительного материала давали всякие книги, ссылки на доп видео, просто заваливали материалами.

Я всегда советую приложение `sololearn`